src/nodes.coffee

```diff
@@ -1272,8 +1273,10 @@ exports.Value = class Value extends Base
     lastProperty instanceof Slice
 
   looksStatic: (className) ->
-    (@this or @base instanceof ThisLiteral or @base.value is className) and
+    return no unless ((thisLiteral = @base) instanceof ThisLiteral or (name = @base).value is className) and
       @properties.length is 1 and @properties[0].name?.value isnt 'prototype'
+    return
+      staticClassName: thisLiteral ? name
```

Comment:
In order to be able to generate AST that include the actual "static class name reference" (eg `A` from `A.b`, `this`, or `@` in the following:
```
class A
  A.b = ->
  @c: 2
  this.d = 3
```
, return that static class name reference as part of the return value from `looksStatic()` (while preserving its existing calling interface which expects a truthy or falsy return value)